src/shared.js

```diff
@@ -31,7 +31,7 @@ module.exports = {
         common: {
           test: /[\\/]node_modules[\\/]/,
           name(module) {
-            const packageName = module.context.match(/[\\/]node_modules[\\/](.*?)([\\/]|$)/)[1];
+            const packageName = module.nameForCondition().match(/[\\/]node_modules[\\/](.*?)([\\/]|$)/)[1];
```

Comment:
I wonder if we can rely on `nameForCondition` existing? In the source code there's always a guard clause against it i.e. https://github.com/webpack/webpack/blob/8522a2d86d316b9a5924c0ca4179af9634a9c44b/lib/optimize/SplitChunksPlugin.js#L330

Reply:
Hmm yeah, I'll stick that in. Thanks @wannabefro 👍

Reply:
> I thought the SplitChunksPlugin was creating the chunks, so which chunks is that code matching against?

@thomasmarshall My understanding is that a chunk is a group of modules that are still being processed by webpack, so SplitChunksPlugin may well be operating on chunks that already exist, doing further processing on them before they are output to bundles.

> Could there still be a case where the regex doesn't match against module.nameForCondition() and instead matches against a chunk.name? 

I think it is possible, but it doesn't look like any of our current chunks would match the regexp. I don't know enough about how chunk names are assigned to say for sure, though.